src/catchup/CatchupWork.cpp

```diff
@@ -403,6 +404,13 @@ CatchupWork::runCatchupStep()
             }
 
             std::vector<std::shared_ptr<BasicWork>> seq;
+            // Before replaying ledgers, ensure Herder state is consistent with
+            // LCL
+            auto consistencyWork =
+                std::make_shared<EnsureHerderConsistencyWork>(
```

Comment:
Mostly for clarity: placing this work in a WorkSequence makes the order of events clear. I can switch to using WorkWithCallback (I was trying to avoid adding more stuff to CatchupWork as it's pretty huge and getting hard to follow)